api/jobs/jobs.py

```diff
@@ -83,10 +125,14 @@ def load(cls, e):
         # Don't modify the map
         d = copy.deepcopy(e)
 
-        if d.get('inputs', None):
-            inputs = d['inputs']
-            for x in inputs.keys():
-                inputs[x] = create_filereference_from_dictionary(inputs[x])
+        if d.get('inputs'):
+            input_dict = {}
+
+            for i in d['inputs']:
```

Comment:
That is true, but if we always interact with jobs through the Job class instead of directly with the mongo object we won't have an issue, as it is a map/dict. Mongo has the option of a unique index key, but it does almost the exact opposite of what we want:

https://docs.mongodb.com/manual/core/index-unique/#unique-constraint-across-separate-documents